### PR TITLE
chapter 12 : Reset Password

### PR DESCRIPTION
**Pull request by daniadore**
Ready

### CODE REVIEW

**Comment by daniadore** [comment]
Ready

File: app/assets/stylesheets/password_resets.scss [hunk 1]
```diff
@@ -0,0 +1,3 @@
+// Place all the styles related to the PasswordResets controller here.
```
Comment (anlvq-1058) [inline app/assets/stylesheets/password_resets.scss:1]: please delete this file

File: app/controllers/password_resets_controller.rb [hunk 2]
```diff
@@ -0,0 +1,57 @@
+class PasswordResetsController < ApplicationController
+  before_action :get_user, only: [:edit, :update]
+  before_action :valid_user, only: [:edit, :update]
+  before_action :check_expiration, only: [:edit, :update]
```
Comment (anlvq-1058) [inline app/controllers/password_resets_controller.rb:4]: combine to 1 line

File: app/views/password_resets/edit.html.erb [hunk 3]
```diff
@@ -0,0 +1,15 @@
+<% provide(:title, 'Reset password') %>
+<h1>Password reset</h1>
```
Comment (anlvq-1058) [inline app/views/password_resets/edit.html.erb:2]: i18n please

File: app/views/user_mailer/password_reset.html.erb [hunk 4]
```diff
@@ -1,5 +1,12 @@
...
+<p>To reset your password click the link below:</p>
+
+  <%= link_to "Reset password", edit_password_reset_url(id: @user.reset_token, email: @user.email) %>
+
+<p>This link will expire in two hours.</p>
 
 <p>
-  <%= @greeting %>, find me in app/views/user_mailer/password_reset.html.erb
+If you did not request your password to be reset, please ignore this email and
+your password will stay as it is.
```
Comment (anlvq-1058) [inline app/views/user_mailer/password_reset.html.erb:11]: indent please

**Review by anlvq-1058 — COMMENTED** [review]
cmt

**Comment by daniadore** [comment]
Ready

File: app/controllers/password_resets_controller.rb [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+      flash[:success] = t(:password_reset)
+      redirect_to @user
+    else
+      render :edit
+    end
+  end
+
+  private
+
+    def check_expiration
```
Comment (ngocvt-0484) [inline app/controllers/password_resets_controller.rb:36]: wrong indent
it should align with private above

File: config/routes.rb [hunk 6]
```diff
@@ -1,4 +1,6 @@
 Rails.application.routes.draw do
+  get 'password_resets/new'
+  get 'password_resets/edit'
```
Comment (ngocvt-0484) [inline config/routes.rb:3]: remove line 2, 3
because you defined routes on line 18

**Comment by daniadore** [comment]
Ready

**Comment by ngocvt-0484** [comment]
merged

**Pull request merged**